src/replayer.py

```diff
@@ -0,0 +1,306 @@
+        '{ "timestamp": "%(asctime)s", "log_level": "%(levelname)s", "message": "%(message)s", '
+        f'"environment": "{args.environment}", "application": "{args.application}", '
+        f'"module": "%(module)s", "process": "%(process)s", '
+        f'"thread": "[%(thread)s]", "hostname": "{hostname}" }} '
+    )
+
+    new_handler.setFormatter(logging.Formatter(json_format))
+    the_logger.addHandler(new_handler)
+    new_level = logging.getLevelName(logger_level.upper())
+    the_logger.setLevel(new_level)
+
+    if the_logger.isEnabledFor(logging.DEBUG):
+        boto3.set_stream_logger()
+        the_logger.debug(f'Using boto3", "version": "{boto3.__version__}')
+
+    return the_logger
+
+
+def get_parameters():
+    parser = argparse.ArgumentParser(
+        description="Convert S3 objects into Kafka messages"
+    )
+
+    # Parse command line inputs and set defaults
+    parser.add_argument("--aws-profile", default="default")
+    parser.add_argument("--aws-region", default="eu-west-2")
+    parser.add_argument("--environment", default="NOT_SET")
+    parser.add_argument("--application", default="NOT_SET")
+    parser.add_argument("--log-level", default="INFO")
+    parser.add_argument("--aws-region", default="eu-west-1")
+    parser.add_argument("--api-hostname", default="NOT_SET")
+
+    _args = parser.parse_args()
+
+    # Override arguments with environment variables where set
+    if "AWS_PROFILE" in os.environ:
+        _args.aws_profile = os.environ["AWS_PROFILE"]
+
+    if "AWS_REGION" in os.environ:
+        _args.aws_region = os.environ["AWS_REGION"]
+
+    if "ENVIRONMENT" in os.environ:
+        _args.environment = os.environ["ENVIRONMENT"]
+
+    if "APPLICATION" in os.environ:
+        _args.application = os.environ["APPLICATION"]
+
+    if "LOG_LEVEL" in os.environ:
+        _args.application = os.environ["LOG_LEVEL"]
+
+    if "API_HOSTNAME" in os.environ:
+        _args.application = os.environ["API_HOSTNAME"]
+
+    required_args = ["API_HOSTNAME"]
+    missing_args = []
+    for required_message_key in required_args:
+        if required_message_key not in _args:
+            missing_args.append(required_message_key)
+    if missing_args:
+        raise argparse.ArgumentError(
+            None,
+            "ArgumentError: The following required arguments are missing: {}".format(
+                ", ".join(missing_args)
+            ),
+        )
+
+    return _args
+
+
+args = get_parameters()
+logger = setup_logging(args.log_level)
+region = os.environ.get("AWS_REGION")
+
+
+def handler(event, context):
+
+    session = boto3.session.Session()
+    default_credentials = session.get_credentials().get_frozen_credentials()
+
+    loaded_event = json.loads(event)
+    try:
+        original_request = loaded_event["Body"]["originalRequest"]
+        original_response = loaded_event["Body"]["originalResponse"]
+    except Exception as e:
+        logger.error("Attempted to extract event items but was unable.")
```

Comment:
currently you are continuing on error
```suggestion
    except KeyError as e:
        logger.error("Attempted to extract event items but was unable.")
        raise e
```